TracingQueryTest flaky on DSE builds

I have seen already a few times the same reason for the failing DSE build:

```
Step #4 - "dse-6.8": INFO  [main] 2021-11-02 10:35:30,917 ExternalStorage.java:311 - Storage cluster (version 6.8) that was requested by [engine:junit-jupiter]/[class:io.stargate.it.http.graphql.cqlfirst.MetricsTest] has been removed.
Step #4 - "dse-6.8": [INFO] 
Step #4 - "dse-6.8": [INFO] Results:
Step #4 - "dse-6.8": [INFO] 
Step #4 - "dse-6.8": [ERROR] Failures: 
Step #4 - "dse-6.8": [ERROR]   TracingQueryTest.tracingIdNormalQueryEnabledGetTracingData:180->validateTrace:214 
Step #4 - "dse-6.8": Expecting:
Step #4 - "dse-6.8":  <0L>
Step #4 - "dse-6.8": to be greater than:
Step #4 - "dse-6.8":  <0L> 
Step #4 - "dse-6.8": [INFO] 
Step #4 - "dse-6.8": [ERROR] Tests run: 1000, Failures: 1, Errors: 0, Skipped: 4
```

Seems that something is wrong here and we should have a look.

--- a/testing/src/main/java/io/stargate/it/grpc/GrpcIntegrationTest.java
+++ b/testing/src/main/java/io/stargate/it/grpc/GrpcIntegrationTest.java
@@ -41,23 +41,25 @@ import io.stargate.proto.StargateGrpc;
 import io.stargate.proto.StargateGrpc.StargateBlockingStub;
 import java.io.IOException;
 import java.util.Arrays;
+import java.util.concurrent.TimeUnit;
 import org.apache.http.HttpStatus;
-import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
 
 @IfBundleAvailable(bundleName = "grpc")
 public class GrpcIntegrationTest extends BaseIntegrationTest {
   private static final ObjectMapper objectMapper = new ObjectMapper();
 
-  protected StargateBlockingStub stub;
-  protected String authToken;
+  protected static ManagedChannel managedChannel;
+  protected static StargateBlockingStub stub;
+  protected static String authToken;
 
-  @BeforeEach
-  public void setup(StargateConnectionInfo cluster) throws IOException {
+  @BeforeAll
+  public static void setup(StargateConnectionInfo cluster) throws IOException {
     String seedAddress = cluster.seedAddress();
 
-    ManagedChannel channel =
-        ManagedChannelBuilder.forAddress(seedAddress, 8090).usePlaintext().build();
-    stub = StargateGrpc.newBlockingStub(channel);
+    managedChannel = ManagedChannelBuilder.forAddress(seedAddress, 8090).usePlaintext().build();
+    stub = StargateGrpc.newBlockingStub(managedChannel);
 
     objectMapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
 
@@ -71,6 +73,22 @@ public class GrpcIntegrationTest extends BaseIntegrationTest {
     AuthTokenResponse authTokenResponse = objectMapper.readValue(body, AuthTokenResponse.class);
     authToken = authTokenResponse.getAuthToken();
     assertThat(authToken).isNotNull();
+  }
+
+  @AfterAll
+  public static void cleanUp() {
+    managedChannel.shutdown();
+    try {
+      if (!managedChannel.awaitTermination(3, TimeUnit.SECONDS)) {
+        managedChannel.shutdownNow();
+        if (!managedChannel.awaitTermination(5, TimeUnit.SECONDS)) {
+          throw new RuntimeException("ManagedChannel failed to terminate, aborting..");
+        }
+      }
+    } catch (InterruptedException ie) {
+      managedChannel.shutdownNow();
+      Thread.currentThread().interrupt();
+    }
   }
 
   protected StargateBlockingStub stubWithCallCredentials(String token) {

--- a/testing/src/main/java/io/stargate/it/grpc/TracingQueryTest.java
+++ b/testing/src/main/java/io/stargate/it/grpc/TracingQueryTest.java
@@ -207,12 +207,15 @@ public class TracingQueryTest extends GrpcIntegrationTest {
     assertThat(traces.getStartedAt()).isGreaterThan(0);
     assertThat(traces.getId()).isNotNull();
 
-    assertThat(traces.getEventsList()).isNotEmpty();
-    QueryOuterClass.Traces.Event event = traces.getEvents(0);
-
-    assertThat(event.getActivity()).isNotEmpty();
-    assertThat(event.getSourceElapsed()).isGreaterThan(0);
-    assertThat(event.getThread()).isNotEmpty();
-    assertThat(event.getSource()).isNotEmpty();
+    assertThat(traces.getEventsList())
+        .isNotEmpty()
+        .allSatisfy(
+            event -> {
+              assertThat(event.getEventId()).isNotEmpty();
+              assertThat(event.getActivity()).isNotEmpty();
+              assertThat(event.getThread()).isNotEmpty();
+              assertThat(event.getSource()).isNotEmpty();
+              assertThat(event.getSourceElapsed()).isGreaterThanOrEqualTo(0);
+            });
   }
 }

